Sources/NIOCore/Codec.swift

```diff
@@ -768,9 +768,11 @@ public final class MessageToByteHandler<Encoder: MessageToByteEncoder>: ChannelO
     private var state: State = .notInChannelYet
     private let encoder: Encoder
     private var buffer: ByteBuffer? = nil
+    private let maxBufferCapacity: Int
 
-    public init(_ encoder: Encoder) {
+    public init(_ encoder: Encoder, maxBufferCapacity: Int = 2048) {
```

Comment:
We can't add a new parameter to this initializer, it's an API break. We need to add a new initializer.

Reply:
Is this the suggestion? I don't see how this is a better alternative to the above. Or are we making a breaking change and forcing consumers to specify this field as part of their implementation?

![Screenshot 2024-06-22 at 2 25 41 pm](https://github.com/apple/swift-nio/assets/47680736/2f22bf79-ab28-4d62-88fb-04ee99724384)

Reply:
The non-breaking method is a new initialiser with a default initialiser, but at that point, why not keep he original implementation. Is there some principle of API creation I am violating?

Reply:
I have pushed something, take a look at LMK

Reply:
The new API shape is right.

Just to clarify, it is always an API break to add a new parameter to a method, even if that parameter has a default value. The _reason_ it's a break is that you can refer to a method by its "fully qualified name": `MessageToByteEncoder.init(_:)`. If you add a new parameter with a default value that method no longer exists. Having two methods as you've written here is the right spelling.

Reply:
Thanks. Had to sleep on that one to understand it.